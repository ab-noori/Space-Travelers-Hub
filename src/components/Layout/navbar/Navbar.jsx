import React from 'react';
import { NavLink } from 'react-router-dom';

import menuItems from './navigationsLinks';
import './style.scss';
import logo from '../../../assets/logo-planet.png';

const Navbar = () => (
  <nav className="navbar">
    <div className="logo">
      <img src={logo} alt="Space Travelers" />
      <h2>Space Travelers Hub</h2>
    </div>
    <ul className="menu-items">
      {menuItems.map((item) => (
        <li key={item.id}>
          <NavLink to={item.path} activeclassname="active" className="menu-link">
            {item.text}
          </NavLink>
        </li>
      ))}
    </ul>
  </nav>
);

export default Navbar;
