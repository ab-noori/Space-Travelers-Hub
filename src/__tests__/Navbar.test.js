import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navbar from '../components/Layout/navbar/Navbar';

test('renders correctly', () => {
  const { container } = render(
    <Router>
      <Navbar />
    </Router>,
  );

  expect(container).toMatchSnapshot();
});
