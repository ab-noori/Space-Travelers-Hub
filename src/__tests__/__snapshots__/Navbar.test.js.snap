// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    class="navbar"
  >
    <div
      class="logo"
    >
      <img
        alt="Space Travelers"
        src="logo.png"
      />
      <h2>
        Space Traverlers' Hub
      </h2>
    </div>
    <ul
      class="menu-items"
    >
      <li>
        <a
          activeclassname="active"
          class="menu-link"
          href="/Space-Travelers-Hub/"
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          aria-current="page"
          class="menu-link active"
          href="/"
        >
           
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          class="menu-link"
          href="/Space-Travelers-Hub/missions"
        >
          Missions
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          aria-current="page"
          class="menu-link active"
          href="/"
        >
          |
        </a>
      </li>
      <li>
        <a
          activeclassname="active"
          class="menu-link"
          href="/Space-Travelers-Hub/profile"
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
